plugins/out_s3/s3.c

```diff
@@ -1186,6 +1215,17 @@ static int put_all_chunks(struct flb_s3 *ctx)
                 return -1;
             }
 
+            if (ctx->compression != FLB_AWS_COMPRESS_NONE) {
+                /* Map payload */
+                ret = flb_aws_compression_compress(ctx->compression, buffer, buffer_size, &payload_buf, &payload_size);
+                if (ret == -1) {
+                    flb_plg_error(ctx->ins, "Failed to compress data, upload UnCompressed data instead");
```

Comment:
language nit: "Failed to compress data, uploading uncompressed data instead to prevent data loss"